src/components/accounts/recovery_setup/SetupRecoveryMethod.js

```diff
@@ -104,13 +124,17 @@ class SetupRecoveryMethod extends Component {
     }
 
     handleSetupRecoveryMethod = (securityCode) => {
-        const  { accountId, setupRecoveryMessage, redirectToApp } = this.props;
+        const  { accountId, setupRecoveryMessage, redirectToApp, url, history } = this.props;
 
         setupRecoveryMessage(accountId, this.method, securityCode)
             .then(({ error }) => {
                 if (error) return;
 
-                redirectToApp();
+                if (url && Object.entries(url).length !== 0) {
```

Comment:
@Patrick1904 any reason not to encapsulate this into action? Seems like get copy-pasted between different places?

Reply:
@vgrichina good point. I added a `fallBack` `url` to `redirectApp`